### PR TITLE
Simplify rejection handler to propagate error by re-throwing reason.

### PR DESCRIPTION
**Pull request by sparhami**

### CODE REVIEW

File: src/event-helper.js [hunk 1]
```diff
@@ -132,6 +132,6 @@ function racePromise_(promise, unlisten, timeout) {
     return result;
   }, (reason) => {
     unlisten();
-    return Promise.reject(reason);
+    throw reason;
```
Comment (dvoytenko) [inline src/event-helper.js]: Do we have a guarantee that `reason` would always be an `Error`?

**Comment by sparhami** [comment]
Interestingly enough, you can throw [anything you want in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw).

**Comment by erwinmombay** [comment]
throwing a `string` is generally bad practice though

**Comment by dvoytenko** [comment]
Yeah, we've been trying to avoid that.

**Comment by sparhami** [comment]
By the same token you shouldn't reject a Promise with something that isn't an `Error`. If you converted the code to async/await, you wouldn't throw a `string`.

**Comment by erwinmombay** [comment]
hmm thats a good point. but i generally think of promise rejection as a way to cancel all the handlers subsequent to the current one and not an analog of actual `error handling`.

**Comment by dvoytenko** [comment]
Agree. Would gladly accept a PR that replaces `reject(string)` with `reject(new Error())`. But I do wonder:
(a) We often not in control of rejection reason, e.g. through some other API.
(b) The worst case, I think, is when someone does `Promise.reject()`, i.e. with reason=undefined.

Any thoughts on these?

**Comment by erwinmombay** [comment]
@sparhami friendly ping to follow up on @dvoytenko

**Comment by sparhami** [comment]
I'm not sure why it matters if you don't have control over the upstream. You can throw `undefined` just fine. This change does not introduce any change in behavior.

I would encourage you to change the upstream to throw an Error instead of a string. It is probably a good idea because then you can tell what actually caused the rejection chain rather than getting a string and having to search for it in the code base.

If you would like, I could create a separate PR for that change.

**Comment by dvoytenko** [comment]
Apologies, missed the last report. This sounds good. LGTM. And we'd definitely would welcome a PR to update rejections to use an exception.

**Pull request merged**